### PR TITLE
Encapsulate health check requests into `Extensions.Health`

### PR DESCRIPTION
**Pull request by joshk**

### CODE REVIEW

**Review by lawik — APPROVED** [review]

File: lib/nerves_hub/extensions/health.ex [hunk 1]
```diff
@@ -95,8 +95,13 @@ defmodule NervesHub.Extensions.Health do
...
 
+  def request_health_check(device) do
+    device_internal_broadcast!(device, "health:check", %{})
+  end
+
   defp device_internal_broadcast!(device, event, payload) do
     topic = "device:#{device.identifier}:extensions"
-    NervesHubWeb.DeviceEndpoint.broadcast_from!(self(), topic, event, payload)
+
+    Phoenix.Channel.Server.broadcast_from(NervesHub.PubSub, self(), topic, event, payload)
```
Comment (nshoes) [inline lib/nerves_hub/extensions/health.ex:105]: Are there any behavior differences between this and `NervesHubWeb.DeviceEndpoint.broadcast_from!/4`?

Comment (joshk) [reply lib/nerves_hub/extensions/health.ex:105]: Nope, `Endpoint`s use `Phoenix.Channel.Server` under the hood. The real issue is that we have two Endpoints, and if you use `DeviceEndpoint` within the web UI (which doesn't start the `DeviceEndpoint`) then it causes an exception.

Comment (nshoes) [reply lib/nerves_hub/extensions/health.ex:105]: Ah, and this publishes to all nodes so the task running the devices endpoint gets it. Got it, thanks!

Comment (joshk) [reply lib/nerves_hub/extensions/health.ex:105]: Kinda sorta. The Endpoint doesn't matter on the receiving end; `Endpoint.broadcast` is just a convenience method, really. It's just that the resolving of the `PubSub` to use, which is defined in the config, isn't setup if the Endpoint isn't started. By using `Phoenix.Channel.Server.broadcast` and passing in `NervesHub.PubSub` (the PubSub used in each of the Endpoint configs) we are working around this problem.

**Pull request merged**